query-graphs-js/src/querygraph.ts

```diff
@@ -748,6 +798,49 @@ function federateSubgraphs(supergraph: Schema, subgraphs: QueryGraph[]): QueryGr
       }
     );
   }
+
+  /**
+   * Handling progressive overrides here. For each progressive @override
+   * application (with a label), we want to update all relevant edges to the
+   * overridden field across all subgraphs with their respective override
+   * condition (the label and a T/F value). The "from" subgraph will have an
+   * override condition of `false`, whereas all other occurrences (all
+   * pertaining specifically to the @shareable interaction) will have an
+   * override condition of `true`.
+   */
+  const allOverrideApplications = schemas
+    .flatMap(s => federationMetadata(s)?.overrideDirective().applications() ?? []);
+
+  for (const application of allOverrideApplications) {
+    const label = application.arguments().label;
+    if (!label) continue;
+
+    const field = application.parent;
+    assert(field instanceof NamedSchemaElement, () => `@override should have been on a field, got ${field}`)
+    const type = field.parent;
+
+    for (const [i, subgraph] of subgraphs.entries()) {
+      const [vertex, ...unexpectedAdditionalVertices] = subgraph.verticesForType(type.name);
+      assert(vertex && unexpectedAdditionalVertices.length === 0, () => `Subgraph ${i} should have exactly one vertex for type ${type.name}`);
```

Comment:
(Somewhat unrelated since this section needs updating, but not every subgraph will have the type.)

Reply:
e44e0846